sdk/include/opentelemetry/sdk/metrics/aggregator/histogram_aggregator.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/metrics/instrument.h"
+#include "opentelemetry/sdk/metrics/aggregator/aggregator.h"
+
+#include <variant>
+#include <vector>
+#include <mutex>
+#include <memory>
+
+namespace metrics_api = opentelemetry::metrics;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+template<class T>
+class HistogramAggregator final : public Aggregator<T>
+{
+    
+public:
+    
+    /**
+     * Constructor for the histogram aggregator.  A sorted vector of boundaries is expected and boundaries are
+     * doubles regardless of the aggregator's templated data type.
+     *
+     * Sum is stored in values_[0]
+     * Count is stored in position_[1]
+     */
+    HistogramAggregator(metrics_api::BoundInstrumentKind kind, std::vector<double> boundaries)
+    {
+        this->kind_ = kind;
+        boundaries_   = boundaries;
```

Comment:
Do we want to validate if the boundaries is monotonic?

Reply:
Added a check which throws an invalid_argument exception if provided boundaries are not sorted